cmd/mayactl/app/command/volume_info.go

```
@@ -0,0 +1,177 @@
+}
+
+// NewCmdVolumeInfo shows info of OpenEBS Volume
+func NewCmdVolumeInfo() *cobra.Command {
+	options := CmdVolumeInfoOptions{}
+	cmd := &cobra.Command{
+		Use:     "info",
+		Short:   "Displays the info of Volume",
+		Long:    volumeInfoCommandHelpText,
+		Example: `mayactl volume info --volname <vol>`,
+		Run: func(cmd *cobra.Command, args []string) {
+			util.CheckErr(options.Validate(cmd), util.Fatal)
+			util.CheckErr(options.RunVolumeInfo(cmd), util.Fatal)
+		},
+	}
+	cmd.Flags().StringVarP(&options.volName, "volname", "n", options.volName,
+		"unique volume name.")
+	return cmd
+}
+
+// Validate verifies the command whether volName is passed or not.
+func (c *CmdVolumeInfoOptions) Validate(cmd *cobra.Command) error {
+	if c.volName == "" {
+		return errors.New("--volname is missing. Please try running [mayactl volume list] to see list of volumes.")
+	}
+	return nil
+}
+
+// TODO : Add more volume information
+// RunVolumeInfo runs info command and make call to DisplayVolumeInfo
+func (c *CmdVolumeInfoOptions) RunVolumeInfo(cmd *cobra.Command) error {
+	annotation := &Annotations{}
+	// GetVolumeAnnotation is called to get the volume controller's info such as
+	// controller's IP, status, iqn, replica IPs etc.
+	err := annotation.GetVolAnnotations(c.volName)
+	if err != nil {
+		return nil
+	}
+	if annotation == nil {
+		fmt.Println("annotation is nil")
+		return nil
+	}
+	if annotation.ControllerStatus != "Running" {
+		fmt.Println("Volume not reachable, ", annotation.ControllerStatus)
+		return nil
+	}
+	// Initiallize an instance of ReplicaCollection, json response recieved from the
+	// controllerIP:9501/v1/replicas is to be parsed into this structure via GetVolumeStats.
+	// An API needs to be passed as argument.
+	collection := client.ReplicaCollection{}
+	controllerClient := client.ControllerClient{}
+	_, err = controllerClient.GetVolumeStats(annotation.ClusterIP+v1.ControllerPort, v1.InfoAPI, &collection)
+	if err != nil {
+		return err
```

We should return an error message here saying .. "Unable to gather volume replica details"

---

GetVolumeStats already prints the error

---

see the link given below for more details
https://github.com/utkarshmani1997/maya/blob/fix-%231209-addinfo/cmd/mayactl/app/command/stats_helper.go#L82

---

ok